testapi.pm

```diff
@@ -905,64 +905,69 @@ sub script_sudo {
-your SUT is offline, pass type_command => 1.
+The script content is based on the variable content of C<current_test_script>
+and is typed or fetched through HTTP depending on various parameters. Typing
+can be forced by passing C<type_command => 1> for example when the SUT does
+not provide a usable network connection.
 
-The default timeout for the script is 30 seconds. If you need more, pass a second parameter.
+The default timeout for the script is based on the default in C<wait_serial>
+and can be tweaked by setting the C<$wait> positional parameter.
 
 =cut
 
 sub script_output {
     my ($current_test_script, $wait, %args) = @_;
-    my $suffix = hashed_string("SO$current_test_script");
+    my $marker = hashed_string("SO$current_test_script");
     # 80 is approximate quantity of chars typed during 'curl' approach
     # if script length is lower there is no point to proceed with more complex solution
     $args{type_command} ||= length($current_test_script) < 80;
+    # Surround the actual script output with special markers so that we can
+    # unambiguously separate the expected output from other content that we
+    # might encounter on the serial device depending on how it is used in the
+    # SUT
+    $current_test_script = "echo $marker\$($current_test_script)";
 
     if (is_serial_terminal) {
-        my $cat = "cat - > /tmp/script$suffix.sh; echo $suffix-\$?-";
+        my $cat = "cat - > /tmp/script$marker.sh; echo $marker-\$?-";
         type_string($cat . "\n");
         wait_serial("$cat", undef, 0, no_regex => 1);
         type_string($current_test_script);
         type_string("\n", terminate_with => 'EOT');
-        wait_serial("$suffix-0-");
+        wait_serial("$marker-0-");
     }
     elsif ($args{type_command}) {
-        my $cat = "cat - > /tmp/script$suffix.sh << 'EOF';\n";
+        my $cat = "cat - > /tmp/script$marker.sh << 'EOF';\n";
         type_string($cat);
         type_string($current_test_script . "\nEOF\n");
     }
     else {
         open my $fh, ">", 'current_script' or croak("Could not open file. $!");
         print $fh $current_test_script;
         close $fh;
-        assert_script_run "curl -f -v " . autoinst_url("/current_script") . " > /tmp/script$suffix.sh";
+        assert_script_run "curl -f -v " . autoinst_url("/current_script") . " > /tmp/script$marker.sh";
         script_run "clear";
     }
 
-    my $run_script = "/tmp/script$suffix.sh ; echo SCRIPT_FINISHED$suffix-\$?-";
+    my $run_script = "/tmp/script$marker.sh ; echo SCRIPT_FINISHED$marker-\$?-";
     if (is_serial_terminal) {
-        type_string("/bin/bash -oe pipefail $run_script\n");
+        type_string("/bin/sh -oe pipefail $run_script\n");
         wait_serial($run_script, undef, 0, no_regex => 1);
     }
     else {
-        type_string "(/bin/bash -eox pipefail $run_script)| tee /dev/$serialdev\n";
+        type_string "(/bin/sh -eox pipefail $run_script)| tee /dev/$serialdev\n";
```

Comment:
busybox sh supports the option - dash does not.

test case: put 'false | cat' in /tmp/sh.
```
> bash  t.sh; echo DONE-$?
DONE-0
> bash -eox pipefail t.sh; echo DONE-$?
+ false
+ cat
DONE-1

(looks the same on the busybox ash)

and on ubuntu:
> sh /tmp/t.sh; echo DONE-$?
DONE-0
> sh -eox pipefail /tmp/t.sh; echo DONE-$?
sh: 0: Illegal option -o pipefail
DONE-2